src/toil/server/wes/tasks.py

```diff
@@ -0,0 +1,279 @@
+        elif self.wf_type == "py":
+            command_args = ["python", workflow_url] + options
+        else:
+            # This shouldn't happen if this was executed from the Toil WES server.
+            raise RuntimeError(f"Unsupported workflow type: {self.wf_type}")
+
+        return command_args
+
+    def call_cmd(self, cmd: Union[List[str], str], cwd: str) -> int:
+        """
+        Calls a command with Popen. Writes stdout, stderr, and the command to
+        separate files. This is a blocking call.
+
+        :returns: The exit code of the command.
+        """
+        stdout_f = os.path.join(self.work_dir, "stdout")
+        stderr_f = os.path.join(self.work_dir, "stderr")
+
+        with open(stdout_f, "w") as stdout, open(stderr_f, "w") as stderr:
+            logger.info(f"Calling: '{' '.join(cmd)}'")
+            process = subprocess.Popen(cmd, stdout=stdout, stderr=stderr, close_fds=True, cwd=cwd)
+
+        try:
+            return process.wait()
+        except (KeyboardInterrupt, SystemExit):
+            process.terminate()
+
+            if process.wait(timeout=5) is None:
+                process.kill()
+
+            logger.info("Child process terminated by interruption.")
+            return 130
+
+    def run(self) -> None:
+        """
+        Construct a command to run a the requested workflow with the options,
+        run it, and deposit the outputs in the output directory.
+        """
+
+        # the task has been picked up by the runner and is currently preparing to run
+        self.set_state("INITIALIZING")
+        commands = self.initialize_run()
+
+        # store the job store location
+        with open(os.path.join(self.work_dir, "job_store"), "w") as f:
+            f.write(self.job_store)
+
+        if self.get_state() in ("CANCELING", "CANCELED"):
+            logger.info("Workflow canceled.")
+            return
+
+        self.set_state("RUNNING")
+        self.write("start_time", get_iso_time())
+        self.write("cmd", " ".join(commands))
+
+        exit_code = self.call_cmd(cmd=commands, cwd=self.exec_dir)
+
+        self.write("end_time", get_iso_time())
+        self.write("exit_code", str(exit_code))
+
+        if exit_code == 0:
+            self.set_state("COMPLETE")
+        # non-zero exit code indicates failure
+        elif exit_code == 130:
+            self.set_state("CANCELED")
+        else:
+            self.set_state("EXECUTOR_ERROR")
+
+    def write_output_files(self) -> None:
+        """
+        Fetch all the files that this workflow generated and output information
+        about them to `outputs.json`.
+        """
+        output_obj = {}
+
+        # TODO: read the STDOUT file if it's a CWL workflow? This will work with all job stores. I think.
+
+        if self.job_store.startswith("file:"):
+            for file in os.listdir(self.out_dir):
+                if file.startswith("out_tmpdir"):
+                    shutil.rmtree(os.path.join(self.out_dir, file))
```

Comment:
I don't think file job stores will all start with `file:`, if we're letting the user specify them, right?

Also, why are we dropping this particular directory from the output? Is it not a file any workflow can intend to output?